storage/src/adb/any/mod.rs

```diff
@@ -166,6 +193,65 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher, T: Translato
         Ok(db)
     }
 
+    /// Initialize an [Any] database in a pruned state.
+    /// Removes all existing persisted data and applies the state given in `cfg`.
+    pub(super) async fn init_pruned(
```

Comment:
nit: I feel like `init_synced` would be more consistent (given the name of the config)?